akka-actor-tests/src/test/scala/akka/actor/CoordinatedShutdownSpec.scala

```diff
@@ -150,8 +149,96 @@ class CoordinatedShutdownSpec
         testActor ! "C"
         Future.successful(Done)
       }
-      Await.result(co.run(UnknownReason), remainingOrDefault)
-      receiveN(4) should ===(List("A", "B", "B", "C"))
+      whenReady(co.run(UnknownReason)) { _ =>
+        receiveN(4) should ===(List("A", "B", "B", "C"))
+      }
+    }
+
+    "cancel tasks" in {
+      import system.dispatcher
+      val phases = Map("a" -> emptyPhase, "b" -> phase("a"))
+      val co = new CoordinatedShutdown(extSys, phases)
+
+      val task0Counter = new AtomicInteger()
+
+      val task0: () => Future[Done] = () =>
+        Future {
+          task0Counter.incrementAndGet()
+          Done
+        }
+
+      co.addCancellableTask("a", "task0-copy0")(task0)
+      co.addCancellableTask("a", "task0-copy1")(task0)
+      val cancellable0 = co.addCancellableTask("a", "task0-copy2")(task0)
+
+      object Task1 { // We can also add a method as a task
+        val counter = new AtomicInteger()
+
+        private def taskMethod(): Future[Done] = Future {
+          counter.incrementAndGet()
+          Done
+        }
+
+        def register(phase: String, name: String): Cancellable = co.addCancellableTask(phase, name)(taskMethod _)
+      }
```

Comment:
This that it is a method doesn't change anything from the point of view of the coordinated shutdown, so unless this will go in docs it's just adding some verbosity that isn't needed.

Reply:
Artifact from previous design iteration. Slated for removal.